python/pysmurf/client/util/smurf_util.py

```diff
@@ -4668,3 +4668,73 @@ def find_probe_tone_gap(self, band, n_tone=2, n_scan=5,
                 plt.close()
 
         return gap_freq
+
+    def set_boxcar_window(self, band, win_len):
+        """
+        Sets boxcar window for streaming data.
+
+        Args
+        ------
+        band : int
+            The band.
+        win_len : int
+            Window length. This should be 2.4 MHz divided by the streaming rate.
+        """
+        coef = 1./win_len
+        coef_fixed = np.round(coef*2**15) # Round our coefficient for Fix16_15 data type
+        self.set_boxcar_filter_coef_fixed(band, coef_fixed)
+
+    def get_filt_gain_boxcar(self, band, win_len):
+        """
+        Calculates the gain in the streaming data introduced by the boxcar filter.
+
+        Args
+        ------
+        band : int
+            The band.
+        win_len : int
+            Window length. This should be 2.4 MHz divided by the streaming rate.
+        """
+        coef_fixed = self.get_boxcar_filter_coef_fixed(band)
+        return (coef_fixed * win_len * 2**-15)
+
+    def program_window(self, band, win_len, window_fn=signal.hamming):
+        """
+        Programs window for streaming data.
+        The max length for the window is 2048 (1.2 kHz minimum streaming rate).
+
+        Args
+        -------
+        band : int
+            The band.
+        win_len : int
+            Window length. This should be 2.4 MHz divided by the streaming rate.
+        window_fn : function, optional, default Hamming
```

Comment:
We should add some description of where we typically get these. Feel free to even just reference this webpage (or something like it) https://docs.scipy.org/doc/scipy-0.19.1/reference/signal.html#window-functions